### DISCUSSION TITLE
Option for generating two thumbnails only

### DISCUSSION THREAD

**Discussion by lorand93** [photoprism/photoprism#4327] [Ideas]
First, I want to express my appreciation for the incredible work you've done with PhotoPrism. It's an amazing tool that has the potential to revolutionize photo management for many users, including myself.

However, I have a major concern that I believe is shared by many others in the community: the generation of 8-11 thumbnails per photo. For those of us hosting PhotoPrism on laptops or other limited storage devices, this is not feasible and leads to unnecessary storage consumption and complexity.

My Request:
I would like to suggest a feature that allows users to generate only a single thumbnail per photo, with customizable compression and pixel size limits. This single thumbnail should be sufficient for most use cases, and providing users the control over its quality and dimensions will make the system much more efficient and user-friendly.

### Benefits of This Feature:

**Reduced Storage Usage:** 
Generating fewer thumbnails can significantly reduce the overall storage requirements for PhotoPrism. Currently, creating 8-11 thumbnails per photo results in a large number of small files. These files not only consume more disk space but also impact the efficiency of storage management. Here's how:

 **To expand on this important topic of DiskSpace Consumption** :
 
- Each additional thumbnail, even if small in size, adds up quickly. For a large photo collection, this can lead to gigabytes of additional storage consumed by thumbnails alone. For example, if you have 10,000 photos and each photo generates 10 thumbnails averaging ~50 KB each, that's an additional 5 GB of storage used just for thumbnails that are not necesssary

- **Allocation Unit Size:**
- Modern file systems use an allocation unit size (also known as cluster size) to manage storage. Each file, no matter how small, occupies at least one allocation unit. If the allocation unit size is 4 KB, and a thumbnail file is 1 KB, it still uses 4 KB of disk space. With numerous small files, this leads to inefficient use of disk space due to the overhead caused by unused space within allocation units.
 
- **File System Performance:**
- Having a large number of small files can degrade file system performance. File systems are optimized for a balance of large and small files, and an excess of small files can lead to increased fragmentation and slower access times. This can be particularly problematic on less powerful hardware where disk I/O performance is already a limiting factor.

- **Backup and Sync Overhead:**
- When backing up or syncing your photo library, the presence of many small thumbnail files increases the total number of files that need to be processed. This can slow down backup and sync operations, consume more bandwidth, and increase the time required to complete these tasks.

**Simplified Configuration**: Users can easily manage and control the quality and size of thumbnails according to their specific needs, without being overwhelmed by multiple sizes.

**Enhanced Performance**: With fewer files to generate and manage, the overall performance of the system can improve, especially on less powerful hardware.

**Wider Adoption**: Simplifying the thumbnail generation process will make PhotoPrism more appealing to a broader audience, leading to increased adoption and support for the project.

I genuinely believe that implementing this feature will make PhotoPrism a more attractive and practical solution for many potential users.

Thank you for your time and dedication to improving PhotoPrism.

Best regards!

**Comment by lorand93** [comment]
To add on this in expectations of responses like "Read our FAQs" yes i've read https://docs.photoprism.app/user-guide/settings/advanced/#which-thumbnails-will-be-generated 

My feature request stands valid 🚀

**Comment by lorand93** [comment]
At least creating one thumbnail for full-screen viewing and one thumbnail for gallery view will drastically improve things. 


Here's a normal use case from a normal day to day user. 

I've got an 8tb external HDD. I use this to store all my photos. 
I want to be able and scroll through my gallery from my phone/tablet/laptop. 
I can use photoprism, but it creates so many thumbnails that my 500gb internal ssd from my 2021 i7 laptop gets filled. 

Ideally I can use my laptop's internal ssd as photoprism's storage for 1 or 2 thumbnails only, and either mount my 8tb hdd in Read Only mode and keep it linked, either totally disconnect the HDD so the storage is offline and I have access to the generated thumbnails.

**Comment by lorand93** [comment]
UP

we need an option for generating two thumbnails, one for gallery grid display and one for displaying it on your phone.

**Comment by lipancs** [comment]
There are many other advanced features that rely on these different types of thumbnails, and if turned off, these features will not work.

**Reply by lastzero** [reply]
Yep, see our FAQ for more information on the thumbnail types and settings:
https://docs.photoprism.app/getting-started/faq/#can-i-skip-creating-thumbnails-completely
